tests/core/test_exp_manager.py

```diff
@@ -0,0 +1,109 @@
+def cleanup_local_folder():
+    # Asserts in fixture are not recommended, but I'd rather stop users from deleting expensive training runs
+    assert not Path("./lightning_logs").exists()
+    assert not Path("./NeMo_experiments").exists()
+
+    yield
+
+    if Path("./lightning_logs").exists():
+        shutil.rmtree('./lightning_logs')
+    if Path("./NeMo_experiments").exists():
+        shutil.rmtree('./NeMo_experiments')
+
+
+class TestExpManager:
+    @pytest.mark.unit
+    def test_omegaconf(self):
+        """Ensure omegaconf raises an error when an unexcepted argument is passed"""
+        with pytest.raises(OmegaConfBaseException):
+            exp_manager(None, {"unused": 1})
+
+    @pytest.mark.unit
+    def test_trainer_default_logger(self, cleanup_local_folder, tmp_path):
+        """ Test that a trainer with logger errors out with a number of arguments. Test that it works with
+        create_tensorboard_logger set to False
+        """
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"root_dir": str(tmp_path)})
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"explicit_log_dir": str(tmp_path)})
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"resume": True})
+
+        # Check that exp_manager uses trainer.logger
+        log_dir = exp_manager(test_trainer, {"create_tensorboard_logger": False, "create_checkpoint_callback": False})
+        assert log_dir.resolve() == Path("./lightning_logs/0").resolve()
+        assert Path("./lightning_logs").exists()
+        assert Path("./lightning_logs/0").exists()
+
+    @pytest.mark.unit
+    def test_checkpoint_configurations(self, cleanup_local_folder):
+        """ Test that trainer creating modelcheckpoint and asking exp_manager to do it too results in errors, but
+        is error free if only one is asked to do so.
+        """
+        disable_tb_logger = {"create_tensorboard_logger": False}
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+        with pytest.raises(CheckpointMisconfigurationError):  # Fails because both try to create modelcheckpoint
+            exp_manager(test_trainer, disable_tb_logger)
+
+        # Should success without error
+        exp_manager(test_trainer, {"create_checkpoint_callback": False}.update(disable_tb_logger))
+
+        test_trainer_2 = pl.Trainer(checkpoint_callback=False)
+        exp_manager(test_trainer_2, disable_tb_logger)  # Should success without error
+
+    @pytest.mark.unit
+    def test_default_log_dir(self, cleanup_local_folder):
+        """Check the default of ./NeMo_experiments/default/datetime works as intended"""
```

Comment:
can we try not to mix two naming conventions when it comes to directories and just stay with one use i.e. `lower_case`?

Reply:
Are you proposing to default to nemo_experiments as opposed to NeMo_experiments?
As I mentioned to Vahid, I don't have strong feelings either for or against.

Reply:
yes, for example, OSX is not handling properly small vs capital letters in dir/filenames, so I generally stick to this rule:
"All the file names are written in lower case for better compatibility with both POSIX and Windows."

https://github.com/opencv/opencv/wiki/Coding_Style_Guide

Reply:
ok that is a valid reason, let's default to nemo_experiments.

Reply:
Changed